### PR TITLE
Cleans up Core FIFO API

### PR DESCRIPTION
**Pull request by vertexmc**
- No longer incrs refcount for each API call
- Loads all syn:fifos at cortex startup

### CODE REVIEW

File: synapse/cores/common.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def __init__(self, link, store, **conf):
 
         logger.debug('Initializing Cortex')
 
+        self.on('node:add', self._onAddFifo, form='syn:fifo')
```
Comment (vEpiphyte) [inline synapse/cores/common.py]: We will also want to lift all the syn:fifo notes ons startup and call .gen() for them

**Comment by vertexmc** [comment]
We should wait and rebase this PR after https://github.com/vertexproject/synapse/pull/618 is merged

File: synapse/tests/test_cortex.py [hunk 2]
```diff
@@ -2849,6 +2849,33 @@ def test_cortex_formtufobytufo(self):
...
+                self.eq(5, fiforef_0._syn_refs)
+
+                # Begin to remove them
+                fiforef_3.fini()
+                self.eq(4, fiforef_0._syn_refs)
+                fiforef_2.fini()
+                self.eq(3, fiforef_0._syn_refs)
+                fiforef_1.fini()
+                self.eq(2, fiforef_0._syn_refs)
+
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py]: We should make sure ``fiforef_0`` is actually fini'd at the end here

Comment (vertexmc) [reply synapse/tests/test_cortex.py]: done

**Pull request merged**